calculators/gas_king_rating

```diff
@@ -0,0 +1,123 @@
+<?php
+
+$GLOBALS['rating_calculators']['pink_color_rating'] = [
+    'description' => '
+        gas king rating
+        only objective: gas as much and as soon as you can
+
+        you start with 2000 rating (compression is activated although maybe not efficient, I was too lazy to see how you did it)
+        you lose 10 rating per minute for each gas not taken
+        you win 1, 2, 3, 4 rating per minute for gas 1, 2, 3, 4
+        ',
+    'default_rating' => 2000,
+    'fields' => [ // always available: game_id, player_id, team, winner_team, status
+        'g.duration',
+        // 'g.sc2_version',
+        // 'g.map',
+        // 'g.date_played',
+        // 'g.region',
+        // 'g.mode',
+        // 'g.players_per_team',
+        // 'g.bunker_destroyed_timings',
+        // 'g.players_count',
+        // 'g.mid_control_first_timing',
+        // 'g.mid_control_timings',
+        // 'g.parsed_in',
+
+        // 'gp.name',
+        // 'gp.clan',
+        // 'gp.color',
+        // 'gp.apm',
+        // 'gp.race',
+        'gp.gas_timings',
+        'gp.gas_count',
+        // 'gp.tier_timings',
+        // 'gp.tier',
+        // 'gp.mineral_value_killed',
+        // 'gp.status_timing',
+        // 'gp.recorder',
+        ],
+    'main' => 'dsr_gas_king_rating__main',
+    ];
+
+function same_sign($a, $b)
+{
+    return ($a >= 0 && $b >= 0) || ($a < 0 && $b < 0);
+}
+
+function dsr_gas_king_rating__main(&$teams)
+{
+    foreach($teams as &$team)
+    {
+        foreach($team as &$player)
+        {
+            $diff = 0;
+            $duration_of_the_game_in_minutes = $teams[0][0]['duration'] / 22.4 / 60;
+
+            if($player['gas_count'] === 0)
+            {
+                $player['rating'] -= 4 * 10 * $duration_of_the_game_in_minutes;
+            }
+
+            else
+            {
+                $min_timings = [1.5, 3, 4.5, 10];
+                $number_of_frames_in_a_min = 22.4 * 60;
```

Comment:
you are about to create an array, and do this calculations more than 12 000 000 times in a row. Sometimes you gotta do what you've gotta do, but in this case you can put such constants right under line 49

Reply:
Should be fixed everywhere